src/components/settings/tabs/plugins/PluginCard.tsx

```diff
@@ -50,15 +59,15 @@ export function PluginCard({ plugin, disabled, onRestartNeeded, onMouseEnter, on
             if (restartNeeded) {
                 // If any dependencies have patches, don't start the plugin yet.
                 settings.enabled = true;
-                onRestartNeeded(plugin.name, "enabled");
+                onRestartNeeded(plugin.name, wasEnabled ? "disabled" : "enabled");
```

Comment:
this is always false here because of the if (!wasEnabled)
```suggestion
                onRestartNeeded(plugin.name, "enabled");
```